x-pack/plugins/security_solution/public/common/components/header_section/index.tsx

```diff
@@ -39,37 +41,47 @@ Header.displayName = 'Header';
 
 export interface HeaderSectionProps extends HeaderProps {
   children?: React.ReactNode;
+  getLensAttributes?: GetLensAttributes;
+  growLeftSplit?: boolean;
   headerFilters?: string | React.ReactNode;
   height?: number;
+  hideSubtitle?: boolean;
   id?: string;
+  inspectMultiple?: boolean;
   isInspectDisabled?: boolean;
+  lensAttributes?: LensAttributes | null;
+  showInspectButton?: boolean;
   split?: boolean;
+  stackByField?: string;
   stackHeader?: boolean;
   subtitle?: string | React.ReactNode;
+  timerange?: { from: string; to: string };
   title: string | React.ReactNode;
   titleSize?: EuiTitleSize;
   tooltip?: string;
-  growLeftSplit?: boolean;
-  inspectMultiple?: boolean;
-  hideSubtitle?: boolean;
 }
 
 const HeaderSectionComponent: React.FC<HeaderSectionProps> = ({
   border,
   children,
+  getLensAttributes,
+  growLeftSplit = true,
   headerFilters,
   height,
+  hideSubtitle = false,
   id,
+  inspectMultiple = false,
   isInspectDisabled,
+  lensAttributes,
+  showInspectButton = true,
   split,
+  stackByField,
   stackHeader,
   subtitle,
+  timerange,
```

Comment:
I have the feeling we are putting some very specific logic in a common component used by a lot of different places. Wouldn't it be better to create another header component (`VisualizationHeaderSection`) that includes these especific visualization actions instead of the inspect button, and leave this one clean?

Reply:
these are a lot of new props though. I'm wondering if it would be cleaner to nest the viz props in a parent prop, ex: `props.vizProps`

Reply:
I see, I found that we can pass children to the HeaderSection (like how we did `stack by field dropdown`), I'll remove `VisualizationActions` from HeaderSection and pass it from `MatrixHistogram`